### ISSUE TITLE
Error upgrading kairos 3.21 to 3.2.2 via k8s

### ISSUE THREAD

**Issue by Itxaka** [kairos-io/kairos#3010]
Hello,

When upgrading within kubernetes from e.g.  22.04-standard-amd64-generic-v3.2.2-k3sv1.30.5-k3s1

the upgrade gives below error and can not be applied 

making a symbolic link works as a workaround

ln -s /etc/os-release /etc/kairos-release
```
+ HOST_DIR=/host
+ '[' '' '!=' true ']'
+ '[' -f /etc/kairos-release ']'
++ source /etc/kairos-release
+++ KAIROS_TARGETARCH=amd64
+++ KAIROS_ID=kairos
+++ KAIROS_VERSION=v3.2.2-v1.31.1-k3s1
+++ KAIROS_VERSION_ID=v3.2.2-v1.31.1-k3s1
+++ KAIROS_PRETTY_NAME='kairos-standard-ubuntu-22.04 v3.2.2-v1.31.1-k3s1'
+++ KAIROS_IMAGE_LABEL=22.04-standard-amd64-generic-v3.2.2-k3sv1.31.1-k3s1
+++ KAIROS_FLAVOR_RELEASE=22.04
+++ KAIROS_MODEL=generic
+++ KAIROS_RELEASE=v3.2.2
+++ KAIROS_BUG_REPORT_URL=https://github.com/kairos-io/kairos/issues
+++ KAIROS_HOME_URL=https://github.com/kairos-io/kairos
+++ KAIROS_SOFTWARE_VERSION_PREFIX=k3s
+++ KAIROS_FLAVOR=ubuntu
+++ KAIROS_REGISTRY_AND_ORG=quay.io/kairos
+++ KAIROS_SOFTWARE_VERSION=v1.31.1+k3s1
+++ KAIROS_NAME=kairos-standard-ubuntu-22.04
+++ KAIROS_ID_LIKE=kairos-standard-ubuntu-22.04
+++ KAIROS_IMAGE_REPO=quay.io/kairos/ubuntu:22.04-standard-amd64-generic-v3.2.2-k3sv1.31.1-k3s1
+++ KAIROS_ARTIFACT=kairos-ubuntu-22.04-standard-amd64-generic-v3.2.2-k3sv1.31.1+k3s1
+++ KAIROS_FAMILY=ubuntu
+++ KAIROS_VARIANT=standard
+++ KAIROS_GITHUB_REPO=kairos-io/kairos
++ echo v3.2.2-v1.31.1-k3s1
+ UPDATE_VERSION=v3.2.2-v1.31.1-k3s1
+ '[' -f /etc/kairos-release ']'
++ source /host/etc/kairos-release
/usr/sbin/suc-upgrade: line 13: /host/etc/kairos-release: No such file or directory
+ CURRENT_VERSION=
+ HOST_DIR=/host
+ '[' '' '!=' true ']'
+ '[' -f /etc/kairos-release ']'
++ source /etc/kairos-release
+++ KAIROS_TARGETARCH=amd64
+++ KAIROS_ID=kairos
+++ KAIROS_VERSION=v3.2.2-v1.31.1-k3s1
+++ KAIROS_VERSION_ID=v3.2.2-v1.31.1-k3s1
+++ KAIROS_PRETTY_NAME='kairos-standard-ubuntu-22.04 v3.2.2-v1.31.1-k3s1'
+++ KAIROS_IMAGE_LABEL=22.04-standard-amd64-generic-v3.2.2-k3sv1.31.1-k3s1
+++ KAIROS_FLAVOR_RELEASE=22.04
+++ KAIROS_MODEL=generic
+++ KAIROS_RELEASE=v3.2.2
+++ KAIROS_BUG_REPORT_URL=https://github.com/kairos-io/kairos/issues
+++ KAIROS_HOME_URL=https://github.com/kairos-io/kairos
+++ KAIROS_SOFTWARE_VERSION_PREFIX=k3s
+++ KAIROS_FLAVOR=ubuntu
+++ KAIROS_REGISTRY_AND_ORG=quay.io/kairos
+++ KAIROS_SOFTWARE_VERSION=v1.31.1+k3s1
+++ KAIROS_NAME=kairos-standard-ubuntu-22.04
+++ KAIROS_ID_LIKE=kairos-standard-ubuntu-22.04
+++ KAIROS_IMAGE_REPO=quay.io/kairos/ubuntu:22.04-standard-amd64-generic-v3.2.2-k3sv1.31.1-k3s1
+++ KAIROS_ARTIFACT=kairos-ubuntu-22.04-standard-amd64-generic-v3.2.2-k3sv1.31.1+k3s1
+++ KAIROS_FAMILY=ubuntu
+++ KAIROS_VARIANT=standard
+++ KAIROS_GITHUB_REPO=kairos-io/kairos
++ echo v3.2.2-v1.31.1-k3s1
+ UPDATE_VERSION=v3.2.2-v1.31.1-k3s1
+ '[' -f /etc/kairos-release ']'
++ source /host/etc/kairos-release
/usr/sbin/suc-upgrade: line 13: /host/etc/kairos-release: No such file or directory
+ CURRENT_VERSION=
```

_Originally posted by @Wolverine65 in https://github.com/kairos-io/kairos/issues/2908#issuecomment-2473326287_

**Comment by Itxaka** [comment]
yeah good catch, this is a wrong check on the script side: https://github.com/kairos-io/packages/pull/1139

Basically the suc-upgrade runs with the upgraded system as root (3.2.2) and the underlying OS under /host (3.2.1)
The current version tries to get the read the info fromt he host, so it sources the `$HOST/etc/os-release` for example, but the check to see if we need to read from kairos-release or os-release is done on the upgraded container, which will always have the kairos-release file.

so the check though the underlying OS had the kairos-release file and tried to read from there

**Comment by Itxaka** [comment]
workaround, run before the upgrade:

```
ln -s /etc/os-release /etc/kairos-release
```

**Comment by Itxaka** [comment]
Now fixes on v3.2.3
